tests/hypervisor/test_esx.py

```diff
@@ -5,40 +5,305 @@
 :caseautomation: Automated
 """
 import pytest
-from virtwho import logger
+from virtwho import REGISTER
+from virtwho import RHEL_COMPOSE
+from virtwho import HYPERVISOR
+from virtwho.configure import VirtwhoHypervisorConfig
 
 
+@pytest.mark.usefixtures('function_virtwho_d_conf_clean')
+@pytest.mark.usefixtures('debug_true')
+@pytest.mark.usefixtures('globalconf_clean')
 class TestEsx:
-    @pytest.mark.tier1
-    def test_hostname_option(self):
-        """Just a demo
+    @pytest.mark.tier2
+    def test_type(self, virtwho, function_hypervisor, esx_assertion):
+        """Test the type= option in /etc/virt-who.d/test_esx.conf
 
-        :title: virt-who: esx: test hostname option
-        :id: fb1f5dec-89c7-41e7-a15b-52b843f6f590
+        :title: virt-who: esx: test type option
+        :id: 0f2fcf4f-ac60-4c3c-905a-fabe22536ab2
         :caseimportance: High
-        :tags: tier1
+        :tags: tier2
         :customerscenario: false
         :upstream: no
         :steps:
-            1.
+            1. Configure type option with invalid value.
+            2. Disable the type option in the config file.
+            3. Create another valid config file
+            4. Update the type option with null value
+
         :expectedresults:
-            1.
+            1. Failed to run the virt-who service, find error messages
+            2. Find error message: Error in libvirt backend
+            3. Find error message, the good config works fine
+            4. The good config works fine
         """
-        logger.info("Succeeded to run the 'test_hostname_option'")
+        # type option is invalid value
+        validate_values = esx_assertion['type']
+        validate_values_list = list(validate_values['invalid'].keys())
```

Comment:
Seems the `validate_values_list` not so readable, could we rename it to 
```
assertion = esx_assertion['type']
assertion_invalid_list = list(assertion['invalid'].keys())
```

Reply:
Updated